controller/rpc/scanner.go

```diff
@@ -256,21 +317,32 @@ func ScanPlatform(scanner string, k8sVersion, ocVersion string, timeout time.Dur
 }
 
 func ScanImage(scanner string, ctx context.Context, req *share.ScanImageRequest) (*share.ScanResult, error) {
+	shouldIncrementTask := true
+	hasAcquiredScanner := false
+
+	// When scanner is empty, it indicates that the CI plugins are calling the scan.
+	// In this case, we set shouldIncrementTask to false to avoid double counting.
 	if scanner == "" {
-		// This happens when called by ci/cd scan from the REST
-		scanner = getAvaliableScanner()
-		if scanner == "" {
-			err := fmt.Errorf("No scanner available")
-			log.WithFields(log.Fields{"error": err}).Error()
-			return nil, err
-		}
+		scanner = ScannerMgr.acquireScannerAvailability()
+		shouldIncrementTask = false
+		hasAcquiredScanner = true
 	}
 
-	client, err := getScannerServiceClient(scanner, true)
+	client, scanner, err := ScannerMgr.getScannerServiceClient(scanner, shouldIncrementTask)
 	if err != nil {
 		return nil, err
 	}
-	defer decScanningCount(scanner)
+
+	// Ensure that resources are properly released or task counts are decremented when the function exits
+	defer func() {
+		if hasAcquiredScanner {
+			// Release the scanner slot back to the availability channel
+			ScannerMgr.releaseScannerAvailability(scanner)
```

Comment:
You should call  `defer` as soon as you have a resource to release later, otherwise the scanner availability would not be increased again if it returns between two calls.